src/main/java/de/srendi/advancedperipherals/common/items/WeakAutomataCore.java

```diff
@@ -76,9 +76,11 @@ public InteractionResult interactLivingEntity(@NotNull ItemStack stack, @NotNull
             return InteractionResult.FAIL;
         }
         String entityType = EntityType.getKey(entity.getType()).toString();
+
         if (AUTOMATA_CORE_REGISTRY.containsKey(entityType)) {
-            CompoundTag tag = stack.getOrCreateTag();
-            CompoundTag consumedData = tag.getCompound(CONSUMER_ENTITY_COMPOUND);
+            CompoundTag consumedData = stack.get(CONSUMED_ENTITY_COMPOUND) == null ? new CompoundTag() : stack.get(CONSUMED_ENTITY_COMPOUND);
+            if (consumedData == null)
```

Comment:
uh so idk if consumedData will ever be null here

Reply:
Yeah it can be - I tried it without the null check 😅 

So if we didn't save a compound tag into that component before, it will return null
In 1.20.4 and below, iirc `getCompound` returned a new Compound if none was found

Reply:
Then what's
```java
consumedData = stack.get(CONSUMED_ENTITY_COMPOUND) == null ? new CompoundTag()
```
used for?